packages/gatsby-remark-embed-snippet/src/index.js

```diff
@@ -31,27 +32,29 @@ const getLanguage = file => {
     : extension.toLowerCase()
 }
 
-module.exports = ({ markdownAST }, { directory } = {}) => {
-  if (!directory) {
-    throw Error(`Required option "directory" not specified`)
-  } else if (!fs.existsSync(directory)) {
-    throw Error(`Invalid directory specified "${directory}"`)
-  } else if (!directory.endsWith(`/`)) {
-    directory += `/`
+module.exports = ({ markdownAST, markdownNode }, { directory } = {}) => {
+  if (directory) {
+    if (!fs.existsSync(directory)) {
+      throw Error(`Invalid directory specified "${directory}"`)
+    } else if (!directory.endsWith(`/`)) {
+      directory += `/`
+    }
   }
 
   visit(markdownAST, `inlineCode`, node => {
     const { value } = node
 
     if (value.startsWith(`embed:`)) {
       const file = value.substr(6)
-      const path = normalizePath(`${directory}${file}`)
+      const snippetPath = directory
```

Comment:
With the above changes, I think we can go use the following code

```suggestion
      const snippetPath = normalizePath(path.join(dirname, file))
```

Reply:
nice! I made changes accordingly